src/ObjLoading/Game/IW5/Leaderboard/JsonLeaderboardDefLoader.cpp

```diff
@@ -0,0 +1,145 @@
+#include "JsonLeaderboardDefLoader.h"
+
+#include "Game/IW5/CommonIW5.h"
+#include "Game/IW5/Leaderboard/JsonLeaderboardDef.h"
+
+#include <format>
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+using namespace nlohmann;
+using namespace IW5;
+
+namespace
+{
+    class JsonLoader
+    {
+    public:
+        JsonLoader(std::istream& stream, MemoryManager& memory)
+            : m_stream(stream),
+              m_memory(memory)
+        {
+        }
+
+        bool Load(LeaderboardDef& leaderboardDef) const
+        {
+            const auto jRoot = json::parse(m_stream);
+            std::string type;
+            unsigned version;
+
+            jRoot.at("_type").get_to(type);
+            jRoot.at("_version").get_to(version);
+
+            if (type != "leaderboard" || version != 1u)
+            {
+                std::cerr << std::format("Tried to load leaderboard \"{}\" but did not find expected type leaderboard of version 1\n", leaderboardDef.name);
+                return false;
+            }
+
+            const auto jLeaderboard = jRoot.get<JsonLeaderboardDef>();
+            return CreateLeaderboardFromJson(jLeaderboard, leaderboardDef);
+        }
+
+    private:
+        static bool CreateTrackTypeFlagsFromJson(const JsonLeaderboardDef& jLeaderboardDef, int& gameFlags)
+        {
+            for (const auto gameFlag : jLeaderboardDef.trackTypes)
+                gameFlags |= gameFlag;
+
+            return true;
+        }
```

Comment:
I think you copied some code here but it doesn't apply here the same way 😅 
The naming is a bit off, its not gameFlags here, so the code is probably from materials.
The other thing is that the enum entries have normal enum values, not the flag values.
So you have to or with `1 << enumValue`, not the enum value.